### PR TITLE
docs: update API docs on navbar behavior

### PR DESCRIPTION
**Pull request by Josh-Cena**
## Related PRs

Documentation for #5072.

### CODE REVIEW

**Comment by Josh-Cena** [comment]
@slorber I'm trying to document this new feature. Are we really going to encourage users to provide the type "dropdown"? To me implicitly specifying dropdowns by an additional items field is concise and introduces no ambiguity.

I'll also modify the content in other Navbar item sections. Arranging fields in tables are much easier to follow than code blocks.

File: website/docs/api/themes/theme-configuration.md [hunk 1]
```diff
@@ -216,26 +235,13 @@ module.exports = {
...
-          // Client-side routing, used for navigating within the website.
-          // The baseUrl will be automatically prepended to this value.
           to: 'docs/introduction',
-          // A full-page navigation, used for navigating outside of the website.
-          // You should only use either `to` or `href`.
-          href: 'https://www.facebook.com',
-          // Prepends the baseUrl to href values.
-          prependBaseUrlToHref: true,
-          // The string to be shown.
+          // Only one of "to" or "href" should be used
```
Comment (slorber) [inline website/docs/api/themes/theme-configuration.md:239]: seems fine to remove as it's already explained in the table

File: website/docs/api/themes/theme-configuration.md [hunk 2]
```diff
@@ -208,6 +208,25 @@ You can add items to the navbar via `themeConfig.navbar.items`.
 
 By default, Navbar items are regular links (internal or external).
 
+React Router should automatically apply active link styling to links, but you can use `activeBasePath` in edge cases. For cases in which a link should be active on several different paths (such as when you have multiple doc folders under the same sidebar), you can use `activeBaseRegex`. `activeBaseRegex` is a more flexible alternative to `activeBasePath` and takes precedence over it -- Docusaurus parses it into a regular expression that is tested against the current URL.
```
Comment (slorber) [inline website/docs/api/themes/theme-configuration.md:211]: What about creating a "navbar link" section?

you'd be able to link to it when mentioning "link-like"

Comment (slorber) [reply website/docs/api/themes/theme-configuration.md:211]: the "Navbar items" section could be a more general presentation of the navbar items system

File: website/docs/api/themes/theme-configuration.md [hunk 3]
```diff
@@ -246,35 +252,53 @@ module.exports = {
...
+Navbar items of the type `dropdown` has the additional `items` field, an inner array of navbar items. The dropdown items cannot be dropdowns themselves. In other words, they can of be one of the following types: `default`, `doc`, `docsVersion`. **This is henceforth referred to as link-like items.**
 
-### Navbar dropdown {#navbar-dropdown}
+Note that the dropdown base item is a link as well, so it is possible be clickable by itself and therefore receive any of the props of a [plain navbar link](#navbar-items).
 
-Navbar items can also be dropdown items by specifying the `items`, an inner array of navbar items.
+:::caution
 
-```js {9-19} title="docusaurus.config.js"
+The `dropdown` type used to be inferred from the existence of `items`. This behavior is deprecated and may stop working in the future. Always explicitly define the item to be of type `dropdown`.
```
Comment (slorber) [inline website/docs/api/themes/theme-configuration.md:263]: I think we can remove this for now.

If we decide to deprecate this we'll just add a Joi warning, but maybe it's more convenient to keep it for now

File: website/docs/api/themes/theme-configuration.md [hunk 4]
```diff
@@ -246,35 +252,53 @@ module.exports = {
 };
 ```
 
-React Router should automatically apply active link styling to links, but you can use `activeBasePath` in edge cases. For cases in which a link should be active on several different paths (such as when you have multiple doc folders under the same sidebar), you can use `activeBaseRegex`. `activeBaseRegex` is a more flexible alternative to `activeBasePath` and takes precedence over it -- Docusaurus parses it into a regular expression that is tested against the current URL.
+### Navbar dropdown {#navbar-dropdown}
 
-Outbound (external) links automatically get `target="_blank" rel="noopener noreferrer"` attributes.
+Navbar items of the type `dropdown` has the additional `items` field, an inner array of navbar items. The dropdown items cannot be dropdowns themselves. In other words, they can of be one of the following types: `default`, `doc`, `docsVersion`. **This is henceforth referred to as link-like items.**
```
Comment (slorber) [inline website/docs/api/themes/theme-configuration.md:257]: What I would do is something like this:

```md

#### Navbar dropdown items

Navbar dropdown items only accept the following "link-like" item types:

- [Navbar link](#navbar-link)
- [Navbar doc](#navbar-doc)
- [Navbar doc vesion](#navbar-doc-version)
```

File: website/docs/api/themes/theme-configuration.md [hunk 5]
```diff
@@ -208,6 +208,25 @@ You can add items to the navbar via `themeConfig.navbar.items`.
...
+
+| Field | Value | Explanation | Required | Default |
+| --- | --- | --- | --- | --- |
+| `label` | `string` | The name to be shown for this item. | Yes | N/A |
+| `to` | `string` | Client-side routing, used for navigating within the website. The baseUrl will be automatically prepended to this value. | Yes | N/A |
+| `href` | `string` | A full-page navigation, used for navigating outside of the website. **Only one of `to` or `href` should be used.** | Yes | N/A |
+| `prependBaseUrlToHref` | `boolean` | Prepends the baseUrl to `href` values. | No | `false` |
+| `position` | `'left'` / `'right'` | The side of the navbar this item should appear on. | No | `'left'` |
+| `activeBasePath` | `string` | To apply the active class styling on all routes starting with this path. This usually isn't necessary. | No | `''` |
+| `activeBaseRegex` | `string` | Alternative to `activeBasePath` if required. | No | `''` |
```
Comment (slorber) [inline website/docs/api/themes/theme-configuration.md:225]: Such default values should be nothing, not '' ?

File: website/docs/api/themes/theme-configuration.md [hunk 6]
```diff
@@ -208,6 +208,25 @@ You can add items to the navbar via `themeConfig.navbar.items`.
...
+React Router should automatically apply active link styling to links, but you can use `activeBasePath` in edge cases. For cases in which a link should be active on several different paths (such as when you have multiple doc folders under the same sidebar), you can use `activeBaseRegex`. `activeBaseRegex` is a more flexible alternative to `activeBasePath` and takes precedence over it -- Docusaurus parses it into a regular expression that is tested against the current URL.
+
+Outbound (external) links automatically get `target="_blank" rel="noopener noreferrer"` attributes.
+
+Accepted fields:
+
+| Field | Value | Explanation | Required | Default |
+| --- | --- | --- | --- | --- |
+| `label` | `string` | The name to be shown for this item. | Yes | N/A |
+| `to` | `string` | Client-side routing, used for navigating within the website. The baseUrl will be automatically prepended to this value. | Yes | N/A |
```
Comment (slorber) [inline website/docs/api/themes/theme-configuration.md:220]: Maybe remove default  value N/A and just leave an empty cell?

File: website/docs/api/themes/theme-configuration.md [hunk 7]
```diff
@@ -208,6 +208,25 @@ You can add items to the navbar via `themeConfig.navbar.items`.
...
+| `href` | `string` | A full-page navigation, used for navigating outside of the website. **Only one of `to` or `href` should be used.** | Yes | N/A |
+| `prependBaseUrlToHref` | `boolean` | Prepends the baseUrl to `href` values. | No | `false` |
+| `position` | `'left'` / `'right'` | The side of the navbar this item should appear on. | No | `'left'` |
```
Comment (slorber) [inline website/docs/api/themes/theme-configuration.md:223]: I think TS notation is quite known now, eventually we'll generate those tables in the future so using `'left' | 'right'` seems appropriate

File: website/docs/api/themes/theme-configuration.md [hunk 8]
```diff
@@ -315,23 +347,31 @@ If you use docs with versioning, this special navbar item type that will render
 
 The user will be able to switch from one version to another, while staying on the same doc (as long as the doc id is constant across versions).
 
-```js {5-8} title="docusaurus.config.js"
+Accepted fields:
+
+| Field | Value | Explanation | Required | Default |
+| --- | --- | --- | --- | --- |
+| `position` | `'left'` / `'right'` | The side of the navbar this item should appear on. | No | `'left'` |
+| `dropdownItemsBefore` | Array of [link-like items](#navbar-dropdown) | Add additional dropdown items at the beginning of the dropdown. | No | `[]` |
```
Comment (slorber) [inline website/docs/api/themes/theme-configuration.md:355]: Also TS notation widely known so maybe we could have this?

[LinkLike](#navbar-dropdown)[]

**Review by slorber — COMMENTED** [review]
Thanks, looks like a great improvement ;)

File: website/docs/api/themes/theme-configuration.md (hunk 1 above)
Comment (Josh-Cena) [reply website/docs/api/themes/theme-configuration.md:239]: I think this improves readability than having a lone commented-out `href`? Readers should be able to comprehend every line on one glance.

File: website/docs/api/themes/theme-configuration.md (hunk 2 above)
Comment (Josh-Cena) [reply website/docs/api/themes/theme-configuration.md:211]: Bonne idée👍

File: website/docs/api/themes/theme-configuration.md [hunk 9]
```diff
@@ -246,38 +242,100 @@ module.exports = {
...
+
 ### Navbar dropdown {#navbar-dropdown}
 
-Navbar items can also be dropdown items by specifying the `items`, an inner array of navbar items.
+Navbar items of the type `dropdown` has the additional `items` field, an inner array of navbar items.
+
+Navbar dropdown items only accept the following **"link-like" item types**:
+
+- [Navbar link](#navbar-link)
+- [Navbar doc](#navbar-doc)
```
Comment (slorber) [inline website/docs/api/themes/theme-configuration.md:298]:
```suggestion
- [Navbar doc link](#navbar-doc-link)
```

File: website/docs/api/themes/theme-configuration.md [hunk 10]
```diff
@@ -315,23 +381,31 @@ If you use docs with versioning, this special navbar item type that will render
...
 The user will be able to switch from one version to another, while staying on the same doc (as long as the doc id is constant across versions).
 
-```js {5-8} title="docusaurus.config.js"
+Accepted fields:
+
+| Field | Value | Explanation | Required | Default |
+| --- | --- | --- | --- | --- |
+| `position` | <code>'left' &#124; 'right'</code> | The side of the navbar this item should appear on. | No | `'left'` |
+| `dropdownItemsBefore` | <code>[LinkLikeItem](#navbar-dropdown)[]</code> | Add additional dropdown items at the beginning of the dropdown. | No | `[]` |
+| `dropdownItemsAfter` | <code>[LinkLikeItem](#navbar-dropdown)[]</code> | Add additional dropdown items at the beginning of the dropdown. | No | `[]` |
```
Comment (slorber) [inline website/docs/api/themes/theme-configuration.md:390]:
```suggestion
| `dropdownItemsAfter` | <code>[LinkLikeItem](#navbar-dropdown)[]</code> | Add additional dropdown items at the end of the dropdown. | No | `[]` |
```

File: website/docs/api/themes/theme-configuration.md [hunk 11]
```diff
@@ -370,25 +452,33 @@ If you use the [i18n feature](../../i18n/i18n-introduction.md), this special nav
...
 The user will be able to switch from one locale to another, while staying on the same page.
 
-```js {5-8} title="docusaurus.config.js"
+Accepted fields:
+
+| Field | Value | Explanation | Required | Default |
+| --- | --- | --- | --- | --- |
+| `position` | <code>'left' &#124; 'right'</code> | The side of the navbar this item should appear on. | No | `'left'` |
+| `dropdownItemsBefore` | <code>[LinkLikeItem](#navbar-dropdown)[]</code> | Add additional dropdown items at the beginning of the dropdown. | No | `[]` |
+| `dropdownItemsAfter` | <code>[LinkLikeItem](#navbar-dropdown)[]</code> | Add additional dropdown items at the beginning of the dropdown. | No | `[]` |
```
Comment (slorber) [inline website/docs/api/themes/theme-configuration.md:461]:
```suggestion
| `dropdownItemsAfter` | <code>[LinkLikeItem](#navbar-dropdown)[]</code> | Add additional dropdown items at the end of the dropdown. | No | `[]` |
```

**Review by slorber — APPROVED** [review]
Thanks a lot, that looks better 👍

**Comment by Josh-Cena** [comment]
Thanks, good catch 😁

**Pull request merged**